torch/csrc/jit/ir.cpp

```diff
@@ -1138,6 +1138,31 @@ void Node::removeAllInputs() {
   inputs_.clear();
 }
 
+void Node::permuteInputs(const std::vector<size_t>& new_order) {
+  schema_ = nullptr;
+  AT_ASSERT(new_order.size() == inputs_.size());
+  std::vector<Value*> new_inputs;
+  new_inputs.reserve(new_order.size());
+  for (size_t i = 0; i < new_order.size(); ++i) {
+    new_inputs.push_back(inputs_[new_order[i]]);
```

Comment:
Please use `.at` on all vector lookups into new_order, otherwise this will segfault when bad indices.